### PR TITLE
Updates to app-arch/upx and app-arch/upx-bin

### PR DESCRIPTION
**Pull request by winterheart**
* Renamed app-arch/upx-ucl to app-arch/upx
* EAPI 7
* Removed old
* Closes: https://bugs.gentoo.org/618432 and https://bugs.gentoo.org/645646

### CODE REVIEW

**Comment by juippis** [comment]
Could this be split into 2 different PRs? pkg move in its own, and version bump in its own (after pkg move is done). It's easy to miss something small otherwise, and fixing and rebasing might become irritating to you. 

Also use 3Q-2019 now. The commit message for pkg move should be something like `app-arch/upx-ucl: move package to app-arch/upx` or `rename as app-arch/upx`. 

As to the version bump PR, feel free to add yourself as a proxy-maintainer if you care about this package :) 
https://wiki.gentoo.org/wiki/Project:Proxy_Maintainers/User_Guide#Proxied_maintainer_in_metadata.xml

Feel free to ping me if you still want to work on this, so we can merge the pkg move asap.

**Comment by Alessandro-Barbieri** [comment]
Does it *really* solve the `-Werror` problem?

**Comment by Alessandro-Barbieri** [comment]
Setting `CXXFLAGS_WERROR=" "` will solve the problem

**Comment by winterheart** [comment]
@juippis , see https://github.com/gentoo/gentoo/pull/12624

**Comment by winterheart** [comment]
@Alessandro-Barbieri , it's here: https://github.com/gentoo/gentoo/pull/12087/commits/4445d1b87fa4ffb97e0f5417de7d6ca596a19abc#diff-770ec90ec853581a87210083b5e2fae5R27

**Comment by winterheart** [comment]
@juippis , renewed this pr

File: app-arch/upx-bin/upx-bin-3.95.ebuild [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+RESTRICT="strip"
+
+RDEPEND="!app-arch/upx"
+
+S="${WORKDIR}"
+
+QA_PREBUILT="/opt/bin/upx"
+
+src_install() {
+	cd ${MY_P}*
```
Comment (juippis) [inline app-arch/upx-bin/upx-bin-3.95.ebuild]: Missing `|| die`.

File: app-arch/upx/upx-3.95.ebuild [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+HOMEPAGE="http://upx.github.io/"
+SRC_URI="https://github.com/upx/upx/releases/download/v${PV}/${P}-src.tar.xz"
+
+LICENSE="GPL-2+ UPX-exception" # Read the exception before applying any patches
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~ppc64 ~sparc ~x86 ~x86-fbsd"
+IUSE=""
+
+RDEPEND=">=dev-libs/ucl-1.03
+	sys-libs/zlib:0
```
Comment (juippis) [inline app-arch/upx/upx-3.95.ebuild]: zlib isn't multi-slotted so this `:0` can be removed.

File: app-arch/upx-bin/upx-bin-3.95.ebuild (hunk 1 above)
Comment (winterheart) [reply app-arch/upx-bin/upx-bin-3.95.ebuild]: Done

File: app-arch/upx/upx-3.95.ebuild (hunk 2 above)
Comment (winterheart) [reply app-arch/upx/upx-3.95.ebuild]: Done.

**Comment by winterheart** [comment]
@juippis fixed

**Comment by juippis** [comment]
Thanks!

**Pull request closed without merging**